src/main/java/com/amazon/opendistroforelasticsearch/sql/query/DefaultQueryAction.java

```diff
@@ -223,15 +235,45 @@ private void setSorts(List<Order> orderBys) {
 
                     request.addSort(
                             SortBuilders
-                                    .scriptSort(new Script(order.getName()), getScriptSortType(order))
-                                    .order(SortOrder.valueOf(order.getType())));
+                                    .scriptSort(new Script(orderByName), getScriptSortType(order))
+                                    .order(sortOrder));
                 } else {
-                    request.addSort(order.getName(), SortOrder.valueOf(order.getType()));
+                    if (orderByName.equals(ScoreSortBuilder.NAME)) {
+                        request.addSort(orderByName, sortOrder);
+                    } else {
+                        FieldSortBuilder fieldSortBuilder;
+                        if (sortBuilderMap.containsKey(orderByName)) {
+                            fieldSortBuilder = sortBuilderMap.get(orderByName);
+                            setSortParmas(fieldSortBuilder, order);
+                        } else {
+                            fieldSortBuilder = SortBuilders.fieldSort(orderByName);
+                            setSortParmas(fieldSortBuilder, order);
+                            sortBuilderMap.put(orderByName, fieldSortBuilder);
+                            request.addSort(fieldSortBuilder);
+                        }
+                    }
                 }
             }
         }
     }
 
+
+    private void setSortParmas(FieldSortBuilder fieldSortBuilder, Order order) {
```

Comment:
setSortParams() instead?

Reply:
Good catch with the typo. Will change it.